FAILED: HandshakeRetrySpec

https://github.com/apache/incubator-pekko/actions/runs/4532716547/jobs/7984495441?pr=275#step:6:3640
```
[03-27 14:07:18.972] [info] - must be retried during handshake-timeout (no message loss) *** FAILED *** (7 seconds, 90 milliseconds)
[03-27 14:07:18.973] [info]   java.lang.AssertionError: assertion failed: timeout (6 seconds) during expectMsg while waiting for hello
[03-27 14:07:18.973] [info]   at scala.Predef$.assert(Predef.scala:279)
[03-27 14:07:18.974] [info]   at org.apache.pekko.testkit.TestKitBase.expectMsg_internal(TestKit.scala:472)
[03-27 14:07:18.974] [info]   at org.apache.pekko.testkit.TestKitBase.expectMsg(TestKit.scala:449)
[03-27 14:07:18.975] [info]   at org.apache.pekko.testkit.TestKitBase.expectMsg$(TestKit.scala:449)
[03-27 14:07:18.975] [info]   at org.apache.pekko.testkit.TestKit.expectMsg(TestKit.scala:984)
[03-27 14:07:18.976] [info]   at org.apache.pekko.remote.artery.HandshakeRetrySpec.$anonfun$new$2(HandshakeRetrySpec.scala:48)
[03-27 14:07:18.976] [info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[03-27 14:07:18.977] [info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[03-27 14:07:18.977] [info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[03-27 14:07:18.978] [info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[03-27 14:07:18.979] [info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[03-27 14:07:18.979] [info]   at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1076)
[03-27 14:07:18.980] [info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[03-27 14:07:18.981] [info]   at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[03-27 14:07:18.981] [info]   at org.apache.pekko.remote.artery.ArteryMultiNodeSpec.withFixture(ArteryMultiNodeSpec.scala:58)
[03-27 14:07:18.982] [info]   at org.scalatest.wordspec.AnyWordSpecLike.invokeWithFixture$1(AnyWordSpecLike.scala:1074)
[03-27 14:07:18.982] [info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTest$1(AnyWordSpecLike.scala:1086)
[03-27 14:07:18.983] [info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[03-27 14:07:18.983] [info]   at org.scalatest.wordspec.AnyWordSpecLike.runTest(AnyWordSpecLike.scala:1086)
[03-27 14:07:18.985] [info]   at org.scalatest.wordspec.AnyWordSpecLike.runTest$(AnyWordSpecLike.scala:1068)
[03-27 14:07:18.985] [info]   at org.apache.pekko.testkit.PekkoSpec.runTest(PekkoSpec.scala:64)
[03-27 14:07:18.986] [info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTests$1(AnyWordSpecLike.scala:1145)
[03-27 14:07:18.986] [info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
[03-27 14:07:18.987] [info]   at scala.collection.immutable.List.foreach(List.scala:333)
[03-27 14:07:18.987] [info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[03-27 14:07:18.988] [info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:390)
[03-27 14:07:18.988] [info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:427)
[03-27 14:07:18.989] [info]   at scala.collection.immutable.List.foreach(List.scala:333)
[03-27 14:07:18.989] [info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[03-27 14:07:18.990] [info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[03-27 14:07:18.990] [info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[03-27 14:07:18.991] [info]   at org.scalatest.wordspec.AnyWordSpecLike.runTests(AnyWordSpecLike.scala:1145)
[03-27 14:07:18.991] [info]   at org.scalatest.wordspec.AnyWordSpecLike.runTests$(AnyWordSpecLike.scala:1144)
[03-27 14:07:18.992] [info]   at org.apache.pekko.testkit.PekkoSpec.runTests(PekkoSpec.scala:64)
[03-27 14:07:18.992] [info]   at org.scalatest.Suite.run(Suite.scala:1112)
[03-27 14:07:18.992] [info]   at org.scalatest.Suite.run$(Suite.scala:1094)
[03-27 14:07:18.993] [info]   at org.apache.pekko.testkit.PekkoSpec.org$scalatest$wordspec$AnyWordSpecLike$$super$run(PekkoSpec.scala:64)
[03-27 14:07:18.993] [info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$run$1(AnyWordSpecLike.scala:1190)
[03-27 14:07:18.994] [info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[03-27 14:07:18.994] [info]   at org.scalatest.wordspec.AnyWordSpecLike.run(AnyWordSpecLike.scala:1190)
[03-27 14:07:18.995] [info]   at org.scalatest.wordspec.AnyWordSpecLike.run$(AnyWordSpecLike.scala:1188)
[03-27 14:07:18.995] [info]   at org.apache.pekko.testkit.PekkoSpec.org$scalatest$BeforeAndAfterAll$$super$run(PekkoSpec.scala:64)
[03-27 14:07:18.996] [info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
[03-27 14:07:18.996] [info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
[03-27 14:07:18.997] [info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[03-27 14:07:18.997] [info]   at org.apache.pekko.testkit.PekkoSpec.run(PekkoSpec.scala:64)
[03-27 14:07:18.998] [info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:318)
[03-27 14:07:18.998] [info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:513)
[03-27 14:07:18.999] [info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)
[03-27 14:07:18.999] [info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[03-27 14:07:19.000] [info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[03-27 14:07:19.000] [info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[03-27 14:07:19.000] [info]   at java.lang.Thread.run(Thread.java:750)
```

--- a/remote/src/test/scala/org/apache/pekko/remote/artery/HandshakeRetrySpec.scala
+++ b/remote/src/test/scala/org/apache/pekko/remote/artery/HandshakeRetrySpec.scala
@@ -45,7 +45,10 @@ class HandshakeRetrySpec extends ArteryMultiNodeSpec(HandshakeRetrySpec.commonCo
         newRemoteSystem(name = Some("systemB"), extraConfig = Some(s"pekko.remote.artery.canonical.port = $portB"))
       systemB.actorOf(TestActors.echoActorProps, "echo")
 
-      expectMsg("hello")
+      // The first message is buffered while the outbound handshake is retried and is only
+      // delivered once the handshake completes, which can take up to handshake-timeout (10s).
+      // The default expect timeout is too tight under CI load, so wait well beyond it.
+      expectMsg(15.seconds, "hello")
 
       sel ! Identify(None)
       val remoteRef = expectMsgType[ActorIdentity].ref.get

